### PR TITLE
ClientChat

### PR DESCRIPTION
**Pull request by danielmejiadev**
- Add the base client.
- Add the resources (Authentication, Users, Chat, Channels)
- Unit test added.

### CODE REVIEW

File: src/client/dto/externalClient.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Dtos
+import { ChannelsResponse } from '../resources/channels/dto/channelsResponse';
+import { UserCredentials, Session } from '../resources/auth/dto/authentication';
+import { UserCreate } from '../resources/users/dto/user.create';
+import { User } from '../resources/users/dto/user';
+import { ChatConnect } from '../resources/chat/dto/chatConnect';
+
```
Comment (MateoOlarteCampo) [inline src/client/dto/externalClient.ts:7]: missing author

File: src/client/index.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// Client
```
Comment (MateoOlarteCampo) [inline src/client/index.ts:1]: missing author

File: src/client/helpers/transformer.decorator.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// Dependencies
```
Comment (MateoOlarteCampo) [inline src/client/helpers/transformer.decorator.ts:1]: missing author

File: src/client/dto/externalClient.ts (hunk 1 above)
Comment (danielmejiadev) [reply src/client/dto/externalClient.ts:7]: Done

File: src/client/index.ts (hunk 2 above)
Comment (danielmejiadev) [reply src/client/index.ts:1]: Done

**Pull request merged**